controllers/bpfman-agent/common.go

```diff
@@ -240,15 +242,57 @@ func interfaceInExcludeList(interfaceSelector *bpfmaniov1alpha1.InterfaceSelecto
 	return false
 }
 
+func setupAllowedInterfacesLists(interfaceSelector *bpfmaniov1alpha1.InterfaceSelector) ([]*regexp.Regexp, []string) {
+	var isRegexp = regexp.MustCompile("^/(.*)/$")
+	var allowedRegexpes []*regexp.Regexp
+	var allowedMatches []string
+
+	for _, definition := range interfaceSelector.InterfacesDiscoveryConfig.AllowedInterfaces {
+		definition = strings.Trim(definition, " ")
+		// the user defined a /regexp/ between slashes: compile and store it as regular expression
+		if sm := isRegexp.FindStringSubmatch(definition); len(sm) > 1 {
+			re, err := regexp.Compile(sm[1])
+			if err != nil {
+				return allowedRegexpes, allowedMatches
+			}
+			allowedRegexpes = append(allowedRegexpes, re)
+		} else {
+			// otherwise, store it as exact match definition
+			allowedMatches = append(allowedMatches, definition)
+		}
+	}
+	return allowedRegexpes, allowedMatches
+}
+
+func interfaceInAllowedList(intf string, allowedRegexpes []*regexp.Regexp, allowedMatches []string) bool {
+	if len(allowedRegexpes) == 0 && len(allowedMatches) == 0 {
+		return true
+	}
+
+	for _, re := range allowedRegexpes {
+		if re.MatchString(intf) {
+			return true
+		}
+	}
+
+	for _, n := range allowedMatches {
+		if n == intf {
+			return true
+		}
+	}
+	return false
+}
+
 func getInterfaces(interfaceSelector *bpfmaniov1alpha1.InterfaceSelector, ourNode *v1.Node, discoveredInterfaces *sync.Map) ([]string, error) {
 	var interfaces []string
 
 	if interfaceSelector.InterfacesDiscoveryConfig != nil && interfaceSelector.InterfacesDiscoveryConfig.InterfaceAutoDiscovery != nil &&
 		*interfaceSelector.InterfacesDiscoveryConfig.InterfaceAutoDiscovery {
```

Comment:
Your changes in this PR lgtm.  However, looking at these lines makes me wonder whether we need `InterfaceAutoDiscovery *bool`.  It seems like we can do without it and the fact that `interfaceSelector.InterfacesDiscoveryConfig != nil` would tell us that it's enabled.  What do you think?

Reply:
what if both lists are nil like excludeInterfaces=nil and AllowedInterfaces=nil and we need to still do autodiscovery won't the bool here will be handy ?

Reply:
if we don't provide anything i.e. have InterfaceDiscoveryConfig we will get validation error 
`* spec.programs[1].tcx.links[0].interfaceSelector: Invalid value: 0: spec.programs[1].tcx.links[0].interfaceSelector in body should have at least 1 properties`
so we need to have one field populated